torch/fx/passes/shape_prop.py

```diff
@@ -28,7 +29,50 @@ class TensorMetadata(NamedTuple):
     is_quantized : bool
     qparams: Dict[str, Any]
 
-def _extract_tensor_metadata(result : torch.Tensor, include_contiguity=True) -> TensorMetadata:
+    # Sparse metadata when layout in [ torch.sparse_coo/csr/csc/bsr/bsc ]
+    #   batch_dim + sparse_dim + dense_dim = ndim = len(shape)
+    #   blocksize for bsr/bsc
+    #   idx_dtype denote the types used for compressed indices and positions
+    layout : torch.layout
+    batch_dim : int
+    sparse_dim : int
+    dense_dim : int
+    blocksize : Optional[Tuple[int, int]]
+    idx_dtype : Optional[torch.dtype]
```

Comment:
I suggest making `idx_dtype` non-Optional as it is always present for sparse tensors.

Reply:
Nvm, idx_dtype is not available for strided tensors.